tests/unit_tests.cpp

```diff
@@ -17,26 +17,65 @@ class PamModelsHillTests : public ::testing::Test
     }
 };
 
-TEST_F(PamModelsHillTests, default_config_factory)
+TEST_F(PamModelsHillTests, check_brents_method)
+{
+    // General test parameters
+    // lamda test function with known roots
+    auto poly = [](double x) { return (x + 3) * (x - 1) * (x - 1); };
+
+    std::function<double(double)> f = poly;
+    double lower = -4.0;
+    double upper = 0.75;
+    double tol = 0.001;
+    unsigned int max_iter = 1000U;
+
+    // Test root search
+    double root = pam_models::hill::brents(f, lower, upper, tol, max_iter);
+    double root_solution = -3.0;
+
+    ASSERT_NEAR(root, root_solution, 0.0001);
+}
+
+TEST_F(PamModelsHillTests, check_json_loader)
 {
-    double a = 0.5;
-    double l_MTC = 0.05;
-    pam_models::hill::from_default_json(a, l_MTC);
+    std::string dir_path = std::filesystem::current_path();
```

Comment:
`dir_path` seems to be unused, it could probably be removed?

Reply:
If I understand it correctly, you use it to generate and later delete a test file in the current directory?
It is better to not touch the working directory to avoid accidentally messing with existing files.  Instead, you can save the file in `/tmp`.  Ideally use [`std::tmpnam`](https://en.cppreference.com/w/cpp/io/c/tmpnam) to generate a unique temporary file name.